packages/shared/src/details-page/datetime.ts

```diff
@@ -217,12 +217,13 @@ export const toISODateString = (date: Date): string =>
   )}`;
 
 export const twentyFourHourTime = (
-  date: Date,
+  dateString: string,
   showSeconds?: boolean
 ): string => {
-  const hours = zeroPad(date.getHours() ?? 0);
-  const minutes = `:${zeroPad(date.getMinutes() ?? 0)}`;
-  const seconds = showSeconds ? `:${zeroPad(date.getSeconds() ?? 0)}` : '';
+  const date = dateString ? new Date(dateString) : null;
```

Comment:
This is not used by Fusion right now, so maybe safe to update.

But, this is a generic util... ideally we should make exceptions and update things under `shared` like this very rarely... and make it a practice not to do it under general cases.
```suggestion
  const date = typeof date === 'string' ? new Date(dateString) : date || null;
```

Reply:
https://github.com/red-hat-storage/odf-console/pull/1489/files#r1689748777

Reply:
Currently this util is used only in this component so it's safe to update.

Reply:
it is still exported from the file, which means anyone using the `odf-shared` library can import it...
of course, we can always check Fusion repo and verify it's safe or not (currently Fusion is not using it), but it's matter of practice that only make breaking changes in `odf-shared` when really needed or in case of rare exceptions (which we can always discuss offline and decide).

One case of "rare exceptions" IMO was latest `OCSStorageClusterModel` to `StorageClusterModel` change, StorageCluster is ODF specific entity, so it's highly unlikely any other repo will have need to import it.

Reply:
@SanjalKatiyar I detected the page was crashing  sometimes (when passed a non-date value), so it's justified.

Reply:
After debugging a bit more, I found that actually this method is receiving either date object (that was being coerced to string when printing) or sometimes a number (thus crashing the page).
So I simplified the fix using `lodash`. PTAL.

Reply:
the alternative I suggested should work well too (it's not a hack)...

but I am okay if we don't want to update it or if there are any concerns with the suggested change...

Reply:
^ my above comment was reply to https://github.com/red-hat-storage/odf-console/pull/1489#discussion_r1689825929